### ISSUE TITLE
Check 049: Limit to official whitespace characters?

### ISSUE THREAD

**Issue by madig** [googlefonts/fontbakery#1903]
## Observed behaviour

Check 049 treats these Unicode values as whitespace:
```
  WHITESPACE_CHARACTERS = [
      0x0009, 0x000A, 0x000B, 0x000C, 0x000D, 0x0020, 0x0085, 0x00A0, 0x1680,
      0x2000, 0x2001, 0x2002, 0x2003, 0x2004, 0x2005, 0x2006, 0x2007, 0x2008,
      0x2009, 0x200A, 0x2028, 0x2029, 0x202F, 0x205F, 0x3000, 0x180E, 0x200B,
      0x2060, 0xFEFF
  ]
```

## Expected behaviour

The Unicode 11.0 [property list](https://www.unicode.org/Public/11.0.0/ucd/PropList.txt) attached a whitespace property ("WS") to these Unicode values:
```
0009..000D    ; White_Space # Cc   [5] <control-0009>..<control-000D>
0020          ; White_Space # Zs       SPACE
0085          ; White_Space # Cc       <control-0085>
00A0          ; White_Space # Zs       NO-BREAK SPACE
1680          ; White_Space # Zs       OGHAM SPACE MARK
2000..200A    ; White_Space # Zs  [11] EN QUAD..HAIR SPACE
2028          ; White_Space # Zl       LINE SEPARATOR
2029          ; White_Space # Zp       PARAGRAPH SEPARATOR
202F          ; White_Space # Zs       NARROW NO-BREAK SPACE
205F          ; White_Space # Zs       MEDIUM MATHEMATICAL SPACE
3000          ; White_Space # Zs       IDEOGRAPHIC SPACE
```

The difference isn't large, but maybe we should only check for official whitespace characters? Can't seem to access that information from Python's `unicodedata` module though 😕

**Comment by anthrotype** [comment]
maybe we can add to our fontTools.unicodedata module.
we already can parse these kind of files
https://github.com/fonttools/fonttools/blob/b38e2bd8acbebe98980f63a5bb490010e0c22134/MetaTools/buildUCD.py#L64

we use that to build the Scripts.txt, ScriptExtensions.txt, and Blocks.txt, which have the same format as PropList.txt.

PR? ;)

**Comment by anthrotype** [comment]
python `str` has a `isspace()` method, but apparently is not in sync with PropList.txt, see https://bugs.python.org/issue18236

n

**Comment by madig** [comment]
> PR? ;)

When I get to it ;)

**Comment by anthrotype** [comment]
I was thinking about how to name the new function to look up the PropList.txt...
I wouldn't call it `property` as it clashes with the built-in function.
Maybe `char_property`.. However I feel that maybe that's a bit too broad.. I mean, `name` is a character property as well, as it has its own function. And `PropList.txt` contains lots of these properties.. Maybe it's better to have a function for each of them (that takes a unicode character and returns True/False), or only for the ones we care about, like `is_white_space`.
Feel free to open an issue over at fonttools, thanks.

**Comment by drj11** [comment]
This check is now `com.google.fonts/check/whitespace_ink`

The list includes whitespace that _should_ have drawings, and also incudes non-whitespace that _should not_ have drawings.

Some observations: The "extras" are
U+180E MONGOLIAN VOWEL SEPARATOR
U+200B ZERO-WIDTH SPACE
U+2060 WORD JOINER
U+FEFF ZERO WIDTH NO-BREAK SPACE (but Byte-Order-Mark in actual use)

These are not whitespace (in the properties sense) but should not have drawings (I'm making an assumption about MONGOLIAN VOWEL SEPARATOR because I don't know about it). So they are appropriate for this test. It's entirely possible there are other char codes like this, that are not whitespace, but have no drawing.

OGHAM SPACE MARK is the other way around. It is a whitespace character that is supposed to have a drawing (at least as far as I understand it, I don't know much about the script).

This test is really about glyphs with no drawings, which may or not be whitespace.

In conclusion:
- Some non-whitespace glyphs should not have drawings (and I think we could test for that, as we do now);
- Some (one!) whitespace glyphs _should_ have drawings. So we should remove OGHAM SPACE MARK from this list.
